File: src/main/org/nlogo/prim/etc/_setcurdir.java
package org.nlogo.prim.etc ;

import org.nlogo.api.LogoException;
import org.nlogo.nvm.Command;
import org.nlogo.nvm.Syntax;

public final strictfp class _setcurdir
	extends Command
{
	@Override
	public void perform( final org.nlogo.nvm.Context context ) throws LogoException
	{
		workspace.fileManager().setPrefix( argEvalString( context , 0 ) ) ;
		context.ip = next ;
	}
	@Override
	public Syntax syntax()
	{
		int[] right = { Syntax.TYPE_STRING } ;
		return Syntax.commandSyntax( right ) ;
	}
}

File: src/main/prim/etc/_setcurdir.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.core.{ Syntax, SyntaxJ }
import org.nlogo.nvm.{ Command, Context }

class _setcurdir extends Command {
  override def syntax =
    SyntaxJ.commandSyntax(Array(Syntax.StringType))
  override def perform(context: Context) {
    workspace.fileManager.setPrefix(argEvalString(context, 0))
    context.ip = next
  }
}
